### PR TITLE
New verifications implementation

### PR DESCRIPTION
**Pull request by daniel33gomez**
In this PR the VerifierInterface was refactored to apply in a better way the facade patter using a new abstraction: ScenarioAttributeVerifier.

Verification like: actors.pos must be in the range (0, 100), weapon.power greather than 0 and scenario.scenes positive was implemented.

### CODE REVIEW

**Review by rubiperucci — APPROVED** [review]
Well done!!

File: src/main/api-utilities/ScenarioAttributeVerifier.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+export class ActionAttributeVerifier extends ScenarioAttributeVerifier {
+
+    check(scenario, action) {
+        var result = this.attributeVerifier.check(action, ['actor', 'action', 'scenes']);
+        if (result == true) result = this.actionVerifier.check(action);
+        if (result == true) result = this.actorVerifier.check(scenario, action.actor);
+        if (result == true) result = this.valueVerifier.check(action.scenes, 0, 9999);
+
```
Comment (JeanPeare) [inline src/main/api-utilities/ScenarioAttributeVerifier.js:43]: Here is an unused space, keep in mind.

But what a nice implementation for the verification process, very good job

File: src/main/api-utilities/ScenarioAttributeVerifier.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+export class WeaponAttributeVerifier extends ScenarioAttributeVerifier {
+
+    check(scenario, weapon) {
+        var result = this.attributeVerifier.check(weapon, ['actor', 'name', 'power', 'xScope', 'yScope']);
+        if (result == true) result = this.attributeVerifier.check(scenario, 'actors');
+        if (result == true) result = this.actorVerifier.check(scenario, weapon.actor);
+        if (result == true) result = this.valueVerifier.check(weapon.xScope, 0, 100);
+        if (result == true) result = this.valueVerifier.check(weapon.yScope, 0, 100);
+        if (result == true) result = this.valueVerifier.check(weapon.power, 0, 9999);
+
```
Comment (JeanPeare) [inline src/main/api-utilities/ScenarioAttributeVerifier.js:57]: same as above

File: src/main/api-utilities/VerifierInterface.js [hunk 3]
```diff
@@ -18,43 +27,49 @@ export class VerifierInterface {
...
 
         case 'actor':
-            return AttributesVerifier.check(obj, ['health', 'name', 'weapon', 'type', 'position']);
+            result = this.actorVerifier.check(obj);
+            if (result == true) {
+                if (typeof obj.weapon === 'object') {
+                    obj.weapon.actor = obj.name;
+                    result = this.weaponVerifier.check({actors: [obj]}, obj.weapon);
+                } else {
+                    result = 'weapon is not an objec';
```
Comment (JeanPeare) [inline src/main/api-utilities/VerifierInterface.js:42]: object*

File: src/main/api-utilities/VerifierInterface.js [hunk 4]
```diff
@@ -18,43 +27,49 @@ export class VerifierInterface {
...
                     if (result == true) {
-                        result = AttributesVerifier.check(actor, ['health', 'name', 'weapon', 'type', 'position']);
-                        if (result == true) result = AttributesVerifier.check(actor.weapon, ['power', 'xScope', 'yScope']);
+                        result = this.actorVerifier.check(actor);
+                        if (result == true) {
+                            if (typeof actor.weapon === 'object') {
+                                actor.weapon.actor = actor.name;
+                                result = this.weaponVerifier.check(this.scenario, actor.weapon);
+                            } else {
+                                result = 'weapon is not an objec';
```
Comment (JeanPeare) [inline src/main/api-utilities/VerifierInterface.js:60]: same as above

File: src/test/api-utilities/ScenarioAttributeVerifier.test.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+    test('returns string when the scenes is a negative number', () => {
+        const scenes = -1;
+        expect(verifier.check(scenes)).toBe('value out of range');
+    });
+
+    test('returns string when the scenes are not defined', () => {
+        const scenes = undefined;
+        expect(verifier.check(scenes)).toBe('value out of range');
+    });
+});
```
Comment (JeanPeare) [inline src/test/api-utilities/ScenarioAttributeVerifier.test.js:222]: Very strong unit test.

File: src/test/api-utilities/VerifierInterface.test.js [hunk 6]
```diff
@@ -299,7 +299,7 @@ describe('Verify that the verifier interface class', () => {
             };
             const verifier = new VerifierInterface(scenario, 'scenario');
 
-            expect(verifier.check()).toBe('power does not exist');
+            expect(verifier.check()).toBe('weapon is not an objec');
```
Comment (JeanPeare) [inline src/test/api-utilities/VerifierInterface.test.js:302]: object* again haha that happens

File: src/test/api-utilities/ScenarioAttributeVerifier.test.js (hunk 5 above)
Comment (daniel33gomez) [reply src/test/api-utilities/ScenarioAttributeVerifier.test.js:222]: Is that good or bad? jajja

**Comment by JeanPeare** [comment]
I'm going to approve the pull request, but change the strings that are misspelled as soon as possible, to keep it clean, very good job.

**Review by JeanPeare — APPROVED** [review]

**Comment by daniel33gomez** [comment]
> I'm going to approve the pull request, but change the strings that are misspelled as soon as possible, to keep it clean, very good job.

I just pushed your suggested changes. Pretty good observations, thank you

**Review by Sam-sta — APPROVED** [review]
Really good implementation, i like this way more than the other. And same as Jean Peare, good tests

**Review by alexeindev — APPROVED** [review]
I like the way how its implemented and very good unit test 👍🏻

**Pull request merged**